Modules/_datetimemodule.c

```diff
@@ -1939,8 +1944,46 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
             ptoappend = PyBytes_AS_STRING(freplacement);
             ntoappend = PyBytes_GET_SIZE(freplacement);
         }
+#ifdef Py_NORMALIZE_CENTURY
+        else if (ch == 'Y' || ch == 'G') {
+            /* 0-pad year with century as necessary */
+            PyObject *year_str;                     /* py string, year */
+            PyObject *year;                         /* py int, year */
+            long year_long;                         /* year as long int */
+            char year_formatted[SIZEOF_LONG*5/2+2]; /* formatted year for %Y/%G */
+
+            year_long = PyLong_AsLong(PyTuple_GET_ITEM(timetuple, 0));
```

Comment:
We try to avoid doing multiple things in one line. This is more readable:

```suggestion
            PyObject *item = PyTuple_GET_ITEM(timetuple, 0);
            long year_long = PyLong_AsLong(item);
```

`year_formatted` can be declared just above L1978. Moreover, `buf` would be a better variable name.

Reply:
Makes sense. Thanks.